pyanaconda/modules/common/task/__init__.py

```diff
@@ -67,3 +69,27 @@ def _callback():
 
     task_proxy.Stopped.connect(_callback)
     task_proxy.Start()
+
+
+def wait_for_task(task_proxy, timeout=math.inf):
+    """Wait for an existing and running task with optional timeout.
+
+    If this returns True and the task is finished, you should call the Finish method of the remote
+    task and handle the remote errors. If this returns False, either
+
+    :param task_proxy: a proxy of the remote task
+    :param float timeout: stop waiting after this time in seconds
+    :return bool: did the task finish or not
+    """
+    start = perf_counter()
+    end = start + timeout
+
+    while task_proxy.IsRunning and perf_counter() <= end:
+        sleep(0.1)
```

Comment:
We use `sleep(1)` for `sync_run_task`. Maybe, we should use it also here, because `task_proxy.IsRunning` is not a cheap operation.

Reply:
1 second seems to be awfully coarse. The distant side of the D-Bus call should be performant enough to make 100000s of calls per second, so the real question is, does D-Bus take tens of ms, so that it's comparable to the wait?

Reply:
A quick and dirty test on my test VM* shows average 3 ms per `IsRunning` call, at 25 iterations. I think that's totally fine - 3% query, 97% sleep.

\* 2 CPU, 2 GB memory - intentionally specced low

Reply:
Since we agreed that 0.1 seconds should be enough, could you open a PR for `sync_run_task` in the upstream, please?